extensions/amp-base-carousel/amp-base-carousel.md

```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
+
+```
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.css">
+```
+
+Fully valid AMP pages use the AMP layout system to infer sizing of elements to create a page structure before downloading any remote resources. However, Bento use imports components into less controlled environments and AMP's layout system is inaccessible.
+
+**Container type**
+
+The `amp-base-carousel` component has a defined layout size type. To ensure the component renders correctly, be sure to apply a size to the component and its immediate children (slides) via `height` and `width` or `aspect-ratio`:
+
+```css
+amp-base-carousel {
+  height: 100px;
+  width: 100%;
+}
+
+amp-base-carousel > * {
+  aspect-ratio: 4/1
+}
+```
+
+### Behavior users should be aware of (optional)
+
+#### RTL
+
+`<amp-base-carousel>` currently requires that you let it know when it is in an
+right-to-left (rtl) context (e.g. Arabic, Hebrew pages). While the carousel will
+generally work without this, there may be a few bugs. You can let the carousel
+know that it should operate as `rtl` as follows:
+
+```html
+<amp-base-carousel dir="rtl" …>
+  …
+</amp-base-carousel>
+```
+
+If the carousel is in a RTL context, and you want the carousel to operate as
+LTR, you can explicitly set the `dir="ltr"` on the carousel.
+
+#### Slide layout
+
+Slides are automatically sized by the carousel when **not** specifying
+`mixed-lengths`. You should give the slides `layout="flex-item"`:
+
+```html
+<amp-base-carousel …>
+  <amp-img layout="flex-item" src="…"></amp-img>
+</amp-base-carousel>
+```
+
+The slides have a default height of `100%` when the carousel is laid out
+horizontally. This can easily be changed with CSS or by using
+`layout="fixed-height"`. When specifying the height, the slide will be
+vertically centered within the carousel.
+
+If you want to horizontally center your slide content, you will want to create a
+wrapping element, and use that to center the content.
+
+#### Number of Visible Slides
+
+When changing the number of visible slides using `visible-slides`, in response
+to a media query, you will likely want to change the aspect ratio of the
+carousel itself to match the new number of visible slides. For example, if you
+want to show three slides at a time with a one by one aspect ratio, you would
+want an aspect ratio of three by one for the carousel itself. Similiarly, with
+four slides at a time you would want an aspect ratio of four by one. In
+addition, when changing `visible-slides`, you likely want to change
+`advance-count`.
+
+```html
+<!-- Using an aspect ratio of 3:2 for the slides in this example. -->
+<amp-base-carousel
+  layout="responsive"
+  width="3"
+  height="1"
+  heights="(min-width: 600px) calc(100% * 4 * 3 / 2), calc(100% * 3 * 3 / 2)"
```

Comment:
Hmm. I think we should continue advising to use CSS for this. E.g. 

```
<style>
  #carousel1 {
    height: calc(100% * 3 * 3 / 2);
  }
  @media (min-width: 600px) {
    #carousel1 {
      height: calc(100% * 4 * 3 / 2);
    }
  }
</style>
```

Of course, that'd split the media queries between the element and the attributes. But IMHO it's better to fix one part of it than both have two parts "broken". We can also communicate later that we are pursuing Web spec update to support attribute media queries as well.

Reply:
To clarify, this section is moved verbatim from its former location in the file, and the code samples here were created with the intention of being used on valid AMP documents. Do you still recommend making the change? Or perhaps duplicating the section so there is one version for valid AMP documents (status quo) and one version with the suggested CSS for Bento documents?

Reply:
If it's for valid AMP, then let's keep as is. Sorry for the confusion. But I think we'll be stumbling into these deltas for some time.